op-node/cmd/genesis/cmd.go

```diff
@@ -154,31 +181,49 @@ var Subcommands = cli.Commands{
 			}
 
 			if l1Deployments != "" {
-				log.Info("L1 deployments", "path", l1Deployments)
 				deployments, err := genesis.NewL1Deployments(l1Deployments)
 				if err != nil {
-					return err
+					return fmt.Errorf("cannot read L1 deployments at %s: %w", l1Deployments, err)
 				}
 				config.SetDeployments(deployments)
 			}
 
-			client, err := ethclient.Dial(ctx.String("l1-rpc"))
-			if err != nil {
-				return fmt.Errorf("cannot dial %s: %w", ctx.String("l1-rpc"), err)
+			var l1StartBlock *types.Block
+			if l1StartBlockPath != "" {
+				if l1StartBlock, err = readBlockJSON(l1StartBlockPath); err != nil {
+					return fmt.Errorf("cannot read L1 starting block at %s: %w", l1StartBlockPath, err)
+				}
 			}
 
-			var l1StartBlock *types.Block
-			if config.L1StartingBlockTag == nil {
-				l1StartBlock, err = client.BlockByNumber(context.Background(), nil)
-				tag := rpc.BlockNumberOrHashWithHash(l1StartBlock.Hash(), true)
-				config.L1StartingBlockTag = (*genesis.MarshalableRPCBlockNumberOrHash)(&tag)
-			} else if config.L1StartingBlockTag.BlockHash != nil {
-				l1StartBlock, err = client.BlockByHash(context.Background(), *config.L1StartingBlockTag.BlockHash)
-			} else if config.L1StartingBlockTag.BlockNumber != nil {
-				l1StartBlock, err = client.BlockByNumber(context.Background(), big.NewInt(config.L1StartingBlockTag.BlockNumber.Int64()))
+			if l1RPC != "" {
+				client, err := ethclient.Dial(ctx.String("l1-rpc"))
```

Comment:
is there a particular reason we use `ctx.String("l1-rpc")` here over using the `l1RPC` var?

Reply:
great point we should use that variable, this is left over from a previous iteration